utils/oscap-info.c

```diff
@@ -89,21 +91,103 @@ static inline void _print_xccdf_status(struct xccdf_status *status, const char *
 	}
 }
 
-static inline void _print_xccdf_profiles(struct xccdf_profile_iterator *prof_it, const char *prefix)
+static void _remove_occurence_of_character_from_string(char *string, char c)
 {
-	printf("%sProfiles:\n", prefix);
+	char *reading_ptr = string, *writing_ptr = string;
+	while (*reading_ptr) {
+		*writing_ptr = *reading_ptr;
+		if (*reading_ptr != c)
+			writing_ptr++;
+		reading_ptr++;
+	}
+	*writing_ptr = '\0';
+}
+
+static void _print_xccdf_profile_default(const struct xccdf_profile *prof, const char *prefix)
+{
+	struct oscap_text_iterator *title_it = xccdf_profile_get_title(prof);
+	char *profile_title = oscap_textlist_get_preferred_plaintext(title_it, NULL);
+	oscap_text_iterator_free(title_it);
+	_remove_occurence_of_character_from_string(profile_title, '\n');
+	printf("%s\tTitle: %s\n", prefix, profile_title);
+	free(profile_title);
+	printf("%s\t\tId: %s%s\n", prefix,
+		xccdf_profile_get_id(prof),
+		xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+}
+
+static void _print_xccdf_profile_verbose(const struct xccdf_profile *prof, const char *prefix)
+{
+	struct oscap_text_iterator *text_it;
+	struct oscap_string_iterator *str_it;
+
+	printf("%sProfile\n", prefix);
+
+	text_it = xccdf_profile_get_title(prof);
+	char *profile_title = oscap_textlist_get_preferred_plaintext(text_it, NULL);
+	oscap_text_iterator_free(text_it);
+	_remove_occurence_of_character_from_string(profile_title, '\n');
+	printf("%s\tTitle: %s\n", prefix, profile_title);
+	free(profile_title);
+
+	printf("%s\tId: %s%s\n", prefix,
+		xccdf_profile_get_id(prof),
+		xccdf_profile_get_abstract(prof) ? " (abstract)" : "");
+
+	text_it = xccdf_profile_get_description(prof);
+	char *profile_description = oscap_textlist_get_preferred_plaintext(text_it, NULL);
+	oscap_text_iterator_free(text_it);
+	_remove_occurence_of_character_from_string(profile_description, '\n');
+	printf("%s\tDescription: %s\n", prefix, profile_description);
+	free(profile_description);
+
+	str_it = xccdf_profile_get_platforms(prof);
+	printf("%s\tPlatforms: ", prefix);
```

Comment:
When I try `oscap info --profile` on  SCAP Security Guide Datastreams, I always have this empty. Is it expected to be empty?

Reply:
It seems that if the platform is empty, it applies to all platforms (which is usually the case). After consulting @dahaic , we have reached conclusion that we don't need platform and description is OK.